specs/Compatibility/RuntimeCompatibilityOptions.md

```diff
@@ -0,0 +1,226 @@
+  </PropertyGroup>
+```
+
+## RuntimeCompatibilityOptions.DisabledChanges
+
+Optional list of specific servicing changes to disable.
+
+If you encounter a problem with a specific change in a servicing update, you can disable just that
+change by adding it to the `DisabledChanges` list before calling `Apply`, or by specifying it in
+your app's project file:
+```xml
+  <PropertyGroup>
+    <WindowsAppSDKDisabledChanges>SampleApiCrashFix, OtherSampleApiCrashFix</WindowsAppSDKDisabledChanges>
+  </PropertyGroup>
+```
+
+The [release notes](https://learn.microsoft.com/windows/apps/windows-app-sdk/stable-channel) for
+the Windows App SDK will list the name of each change that can be disabled.
+
+Note that disabling a change is a temporary measure which gives time for a fix to be released in a
+future Windows App SDK update or for you to implement an update in your app. You should report
+changes you disable if you believe it to be a bug in the Windows App SDK. This will help the
+Windows App SDK team to understand the impact of the change and prioritize a fix. You can report
+issues on the [WindowsAppSDK GitHub repo](https://github.com/microsoft/WindowsAppSDK/issues/new?template=bug-report.yaml).
+
+The capability to disable a change is **not** available across new Windows App SDK stable
+releases (such as 1.7.x to 1.8.x), so this does not allow you to permanently disable an
+intentional behavior change. Also, new stable releases, such as 1.8.0, start with no changes
+available to disable -- this capability only applies to disabling targeted servicing changes
+added in servicing releases.
+
+## RuntimeCompatibilityOptions.Apply
+
+Apply the set compatibility options to the runtime.
+
+RuntimeCompatibilityOptions must be applied early in the process before any other Windows App
+SDK APIs are called, or right after initializing the Windows App Runtime. The options must be set
+early before the configuration is locked for the lifetime of the process. Since the Windows App
+Runtime needs to run with a consistent configuration, it will lock the configuration when it needs
+to ensure the configuration will no longer change. Calling `Apply` will also lock the
+configuration.
+
+If the configuration has already been locked, calling `Apply` will throw an `E_ILLEGAL_STATE_CHANGE`
+exception if the options differ from the locked configuration. It is okay to call `Apply` multiple
+times with the same configuration, such as if the process had initialized and used the Windows
+App Runtime earlier and is reinitializing for additional use.
+
+# API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.Windows.ApplicationModel.WindowsAppRuntime
+{
+    /// The set of servicing changes that can be disabled.
+    enum RuntimeCompatibilityChange
+    {
+        None = 0, /// do not use this value
+
+        // Add real changes here:
+
+        // 1.7.1
+        // SampleApiCrashFix,
+        // OtherSampleApiCrashFix,
```

Comment:
Can you give a more realistic example of an enum name?

For example, suppose we made a change to reset the polarity of a widget when it changes. What would we call the enum that suppresses the auto-reset?

Suppose we made a change to fix a crash when a gadget is shown while it is disabled. What would we call the enum that re-enables the crash?

Reply:
For those examples, I'd propose these names:
* WidgetPolarityReset
* ShowDisabledGadgetCrash

Using the list of fixes in 1.6.4, we might have used these names:
* MultiLineTextBoxSelection
* DDLMPackageInstall
* MissingDetoursExport
* CommandBarMenuPositioning
* OldGraphicsHardwareCrash
* PointerEventWindowCloseCrash
* UIAInitReentrancyCrash
* CompositionCapabilitiesCrash
* ExpandedUnicodeSupportForTextBoxes
* PackageManagerRegisterNewerHandling

We may need a separate discussion on the naming scheme to use.